evap/staff/templates/staff_evaluation_operation.html

```diff
@@ -53,11 +53,59 @@
                 {% endif %}
             </div>
         {% endif %}
-        <input type="hidden" name="target_state" value="{{ target_state }}" />
+        <input type="hidden" name="target_state" value="{{ target_state }}"/>
+
+        {% if show_delete_answers_checkbox %}
+            <div class="card mb-3">
+                <div class="card-body">
+                    <div class="form-check">
+                        <input class="form-check-input" type="checkbox" name="delete-previous-answers" id="delete-previous-answers" checked>
+                        <label class="form-check-label" for="delete-previous-answers">
+                            {% trans "Delete previous answers" %}
+                        </label>
+                    </div>
+                </div>
+            </div>
+        {% endif %}
 
         <div class="card card-submit-area card-submit-area-2 text-center mb-3">
             <div class="card-body">
-                <button type="submit" class="btn btn-primary">{% translate 'Confirm' %}</button>
+            {% if show_delete_answers_checkbox %}
+                <confirmation-modal type="submit" form="evaluation-operation-form" id="confirmation-modal">
+                    <span slot="title">{% translate 'Reset Evaluation' %}</span>
+                    <button slot="submit-group" class="btn ms-2 btn-danger" data-event-type="confirm">
+                        {% translate 'Reset Evaluation' %}
+                    </button>
+                    <span slot="question">
+                        {% blocktranslate trimmed %}
+                            This will delete all previously received answers. Voters will have to resubmit the
+                            evaluation.
+                        {% endblocktranslate %}
+                    </span>
+                    {% comment %}
+                        The modal needs an element in the slot, otherwise it throws an error.
+                        No button is needed, as the modal is conditionally shown via JS.
+                    {% endcomment %}
+                    <span slot="show-button"></span>
+                </confirmation-modal>
+                <button id="form-confirmation-btn" type="button" class="btn btn-primary">{% translate 'Confirm' %}</button>
+                <script>
+                    /** @type ConfirmationModal */
+                    const modal = document.getElementById("confirmation-modal");
+                    /** @type HTMLFormElement */
+                    const form = document.getElementById("evaluation-operation-form");
+
+                    document.getElementById("form-confirmation-btn").addEventListener("click", () => {
```

Comment:
I think we can clean this up a bit, I imagine the following structure:

```html
<form id="the-form">
    {% if something %}
        <input type="checkbox" name="delete-previous-answers" checked />
    {% endif %}

    <confirmation-modal type="submit" data-allow-answer-deletion ...>
        ...
    </confirmation-modal>

    <button type="submit" ...></button>
</form>

<script>
    const form = document.getElementById("the-form");
    form.addEventListener("submit", event => {
        const data = new FormData(form);
        if (data.get("delete-previous-answers") && !event.submitter?.hasAttribute("data-allow-answer-deletion")) {
            event.preventDefault();
            form.querySelector("confirmation-modal[data-allow-answer-deletion]").dialog.showModal();
        }
    });
</script>
```

This way, we avoid more `id`s and don't need to include different buttons depending on what value `show_delete_answers_checkbox` has. What do you think?

Reply:
@niklasmohrin I changed it to this. We need to put some kind of marker on the button inside the modal as this element will be used the `.submitter` value.

```html
<form id="evaluation-operation-form">
	<confirmation-modal type="submit" data-allow-answer-deletion>
	...
	    <button slot="submit-group" class="btn ms-2 btn-danger" data-event-type="confirm" data-confirm-answer-deletion>
		{% translate 'Reset Evaluation' %}
	    </button>

	    <span slot="show-button"></span>
	</confirmation-modal>

	<button type="submit" class="btn btn-primary">{% translate 'Confirm' %}</button>
	<script>
	    document.getElementById("evaluation-operation-form").addEventListener("submit", submitEvent => {
		const data = new FormData(submitEvent.target);
		if (data.get("delete-previous-answers") === "on" && !submitEvent.submitter?.hasAttribute("data-confirm-answer-deletion")) {
		    submitEvent.preventDefault()
		    submitEvent.target.querySelector("confirmation-modal[data-allow-answer-deletion]").dialog.showModal();
		}
	    })
	</script>
</form>
```

But I am not sure if I like it, because now the modal and javascript is used on every evaluation operation.

Reply:
Ough I forgot about this: https://github.com/e-valuation/EvaP/blob/a9e417180b6a8e380bbcb9885b301f13e9524bc0/evap/static/ts/src/confirmation-modal.ts#L53-L55
(https://github.com/WICG/webcomponents/issues/814)

I wrote the above thinking that the modal itself would be the submitter :/ 

> But I am not sure if I like it, because now the modal and javascript is used on every evaluation operation.

I am not sure I understand this point - the javascript runs on every submit, but that is not noticable if the condition in the handler is `false`. I prefer handling things in the `submit` events over `click` events because it marks the form as the central actor of the process. The modal is only shown if the delete-answers checkbox is checked, so the modal should only get involved when we really want to ask for additional confirmation - or does it appear more often than needed now? Or am I not getting something else here?

---

I think your solution with the explicit `submit-group` is the only correct way if we want a solution where all _allowed_ submitters are marked with a data attribute. We could also do it the other way around and mark the other submit button as (for example) `data-prompt-confirmation-for-delete-answers` - and we leave the modal just as is. This approach is slightly less nice beacuse the default submission is "unsafe". However, there are only two possible submission sources for the operation form, so this should be fine.

What do you think?

Reply:
re 2: Yeah, I would also go for the current way. It just seems more explicit & readable that way.

re 1: My point was that this way we always send and execute additional javascript to the client, when it is only used for one of the operations. But this was only a minor thought.